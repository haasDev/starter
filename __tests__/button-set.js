import React from 'react';
import ButtonSet from '../src/button-set';
import {mount} from 'enzyme';
import snapshot from './snapshot';

describe('ButtonSet', () => {
  it('should render', () => {
    const buttons = [
      {
        text: 'Save',
        kind: 'primary',
        onClick: () => console.log('saved!'),
      },
      {
        text: 'Info',
        kind: 'info',
        onClick: () => console.log('info!'),
      },
      {
        text: 'Cancel',
        kind: 'danger',
        onClick: () => console.log('cancelled!'),
      },
    ];

    snapshot(<ButtonSet buttons={buttons} />);
  });

  it('should add class = its kind property', () => {
    const buttons = [
      {
        text: 'Text',
        kind: 'success',
        onClick: () => console.log('success!')
      }
    ];
    const wrapper = mount(<ButtonSet buttons={buttons} />);
    const btn = wrapper.find('.btn-success');
    expect(btn.length).toBe(1);
  });

  it('should display the text property as the button label', () => {
    const buttons = [
      {
        text: 'Text',
        kind: 'success',
        onClick: () => console.log('success!')
      }
    ];
    const wrapper = mount(<ButtonSet buttons={buttons} />);
    const btn = wrapper.find('.btn-success');
    expect(btn.text()).toBe('Text');
  });

  it('should call onClick fn when clicked', () => {
    const onClick = jest.fn();
    const buttons = [
      {
        text: 'Text',
        kind: 'success',
        onClick
      }
    ];
    const wrapper = mount(<ButtonSet buttons={buttons} />);
    const btn = wrapper.find('.btn-success');
    btn.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('button should be able to be disabled', () => {
    const buttons = [
      {
        disabled: true,
        text: 'Text',
        kind: 'success',
        onClick: jest.fn
      }
    ];

    const wrapper = mount(<ButtonSet buttons={buttons} />);
    const disabled = wrapper.find('button[disabled=true]');
    expect(disabled.length).toBe(1);
  });
});

