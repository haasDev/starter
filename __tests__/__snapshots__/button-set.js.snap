exports[`DataEntry should render 1`] = `
<div
  className="button-set">
  <button
    className="btn btn-primary"
    onClick={[Function]}>
    Save
  </button>
  <button
    className="btn btn-info"
    onClick={[Function]}>
    Info
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}>
    Cancel
  </button>
</div>
`;
